indigo_app/templates/indigo_api/work_popup.html

```diff
@@ -4,32 +4,55 @@ <h6>
     {% if work.stub %}
       <span class="badge badge-info">stub</span>
     {% endif %}
+    <br/><span class="text-muted mt-1">{{ work.frbr_uri }}</span>
   </h6>
-  <p class="text-muted mb-1 wb-all">{{ work.frbr_uri }}</p>
 
   {% if work.assent_date %}
-    <p class="mb-2">Assented to on {{ work.assent_date|date:"Y-m-d" }}.</p>
+    <p class="mb-2">Assented to on {{ work.assent_date|date:"Y-m-d" }}</p>
   {% endif %}
 
-  <p class="mb-2">Published on {{ work.publication_date|date:"Y-m-d" }}
-    {% if work.publication_name and work.publication_number and work.publication_document %}
-      in <a href="{% url 'work_publication_document' frbr_uri=work.frbr_uri filename=work.publication_document.filename %}" target="_blank" rel="noopener">
-        {{ work.publication_name }} no. {{ work.publication_number }}</a>.
-    {% elif work.publication_name and work.publication_number %}
-      in {{ work.publication_name }} no. {{ work.publication_number }}.
-    {% endif %}
-  </p>
+  {% if work.publication_name or work.publication_date %}
+    <p class="mb-2">
+      Published
+        {% if work.publication_date %}
+          on {{ work.publication_date|date:"Y-m-d" }}
+        {% endif %}
+
+        {% if work.publication_name %}
+          in
+
+          {% if work.publication_document %}
+            <a href="{% url 'work_publication_document' frbr_uri=work.frbr_uri filename=work.publication_document.filename %}" target="_blank" rel="noopener">{{ work.publication_name }}
+            {% if work.publication_number %}
+                no. {{ work.publication_number }}
+            {% endif %}
+            </a>
+
+          {% else %}
+            {{ work.publication_name }}
+              {% if work.publication_number %}
+                no. {{ work.publication_number }}
+              {% endif %}
+          {% endif %}
+
+        {% elif work.publication_document %}
+          <br/><a href="{% url 'work_publication_document' frbr_uri=work.frbr_uri filename=work.publication_document.filename %}" target="_blank" rel="noopener">Publication document</a>
+        {% endif %}
+    </p>
+  {% elif work.publication_document %}
+    <p class="mb-2"><a href="{% url 'work_publication_document' frbr_uri=work.frbr_uri filename=work.publication_document.filename %}" target="_blank" rel="noopener">Publication document</a></p>
+  {% endif %}
 
   {% if work.commencement_date %}
     <p class="mb-2">Commenced on {{ work.commencement_date|date:"Y-m-d" }}{% if work.commencing_work %} by
       <a href="{% url 'work' frbr_uri=work.commencing_work.frbr_uri %}">
-        {{ work.commencing_work.title }}</a>{% endif %}.
+        {{ work.commencing_work.title }}</a>{% endif %}
```

Comment:
I'd like to keep the full stops. They were added because it makes it clearer when the sentence has wrapped or not, since there are multiple lines in the popup.